v2/federation/watch.go

```diff
@@ -20,16 +26,50 @@ type BundleWatcher interface {
 
 	// OnUpdate is called when a bundle has been updated. If a bundle is
 	// fetched but has not changed from the previously fetched bundle, OnUpdate
-	// will not be called.
+	// will not be called. This function is called synchronously from WatchBundle
+	// so all the time this function takes to finish is added to the time taken
+	// by fetching the bundle.
 	OnUpdate(*spiffebundle.Bundle)
 
 	// OnError is called if there is an error fetching the bundle from the
-	// endpoint.
+	// endpoint. This function is called synchronously from WatchBundle
+	// so all the time this function takes to finish is added to the time taken
+	// by fetching the bundle.
```

Comment:
```suggestion
	// endpoint. This function is executed synchronously by WatchBundle and therefore
	// should have a short execution time to prevent blocking the watch.
```